pkgs/development/compilers/go/1.18.nix

```diff
@@ -245,21 +125,15 @@ stdenv.mkDerivation rec {
     export CC=${buildPackages.stdenv.cc}/bin/cc
     ''}
     ulimit -a
-  '';
 
-  postBuild = ''
-    (cd src && ./make.bash)
+    pushd src
+    ./make.bash
+    popd
+    runHook postBuild
   '';
 
-  doCheck = stdenv.hostPlatform == stdenv.targetPlatform && !stdenv.isDarwin;
-
-  checkPhase = ''
```

Comment:
Could we have just a smoke test here, that builds a binary and runs it? Or do you think this is unnecessary because the go compiler itself also builds and run?

Reply:
I'll add a smoke test to https://github.com/NixOS/nixpkgs/pull/185604

Reply:
Ok. Something that requires cgo might be nice to have, to make it a bit less trivial.